File: ApiDart-master/app/Models/DiaDanh.php
<?php

namespace App\Models;
use App\Models\HinhAnhDiaDanh;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class DiaDanh extends Model
{
  use HasFactory;
    protected $table='diadanh';
    protected $fillable = [
      'TenDiaDanh',
      'ViTri',
      'MoTa',
      'HinhAnh',
  ];

  public function Share()
    {
        return $this->hasMany(Share::class, 'DiaDanhId');
    }
}

File: lib/Model/dia_danh.dart
class DiaDanh {
  int? id;
  String? tenDiaDanh;
  String? viTri;
  String? moTa;
  String? hinhAnh;

  DiaDanh({
    this.id,
    this.tenDiaDanh,
    this.viTri,
    this.moTa,
    this.hinhAnh,
  });

  DiaDanh.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    tenDiaDanh = json['TenDiaDanh'];
    viTri = json['ViTri'];
    moTa = json['MoTa'];
    hinhAnh = json['HinhAnhId'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['TenDiaDanh'] = this.tenDiaDanh;
    data['ViTri'] = this.viTri;
    data['MoTa'] = this.moTa;
    data['HinhAnh'] = this.hinhAnh;
    return data;
  }
}
